secp256k1-sys/build.rs

```diff
@@ -20,11 +20,16 @@ fn main() {
                .include("depend/secp256k1/include")
                .include("depend/secp256k1/src")
                .flag_if_supported("-Wno-unused-function") // some ecmult stuff is defined but not used upstream
+               .flag_if_supported("-Wno-unused-parameter") // patching out printf causes this warning
                .define("SECP256K1_API", Some(""))
                .define("ENABLE_MODULE_ECDH", Some("1"))
                .define("ENABLE_MODULE_SCHNORRSIG", Some("1"))
                .define("ENABLE_MODULE_EXTRAKEYS", Some("1"))
-               .define("ENABLE_MODULE_ELLSWIFT", Some("1"));
+               .define("ENABLE_MODULE_ELLSWIFT", Some("1"))
+               // upstream sometimes introduces calls to printf, which we cannot compile
+               // with WASM due to its lack of libc. printf is never necessary and we can
+               // just #define it away.
+               .define("printf(...)", Some(""));
```

Comment:
I think we should only use this if the target is wasm ie., put it in the 
```rust
    // WASM headers and size/align defines.
    if env::var("CARGO_CFG_TARGET_ARCH").unwrap() == "wasm32" {
```

Reply:
Also the `-Wno-unused-paramater`

Reply:
Why?

Reply:
Because there is a bug in `cc::Build::flag_if_supported` so we are getting loads of warnings. Well on my machine anyways.

Reply:
> Why?

Did you  mean why do I think we should put the `.define` on the wasm-only `cc::Build` or why the "Also `the -Wno-unused-parameter`? Ignore the second one, and for the first, I thought this because the code comment specifically mentions that this is a wasm problem so it seems logical to put it in the wasm-only place.

Reply:
I don't think it's wasm-only. It would be any linker that tries to resolve libc symbols before pruning unused code.